### DISCUSSION TITLE
Making empty transparent PNG compatible with Photoshop

### DISCUSSION THREAD

**Discussion by Build-0-Matic** [ImageMagick/ImageMagick#5387] [Help] [answered]
I'm having trouble with something that seems trivial.
How can I make a fully transparent PNG file that will actually be compatible with Photoshop?
I've tried
`magick convert -format png24 -size 450x36 canvas:transparent logo.png`
but Magick crashes.
I tried
`magick convert -format png24 -size 450x36 canvas:none logo.png`
but Photoshop refuses to open the file
I also tried a bunch of other things to no avail. And when I manage to make a file that Photoshop will open, it's a 1bit file that's all black and I have to do various things to get it to become an actual RGB file... annoying

Here's what I'm running
ImageMagick 7.1.0-43 Q16-HDRI x64 c95ef31:20220709 on Windows 10

Thanks!

**Comment by snibgo** [comment, marked as answer]
"`-format png24`" doesn't do what you think it does. Instead, I suggest "`PNG32:`" prefix on the output filename, ie "`PNG32:logo.png`".

You need 32 not 24, because you want to record the alpha channel in the file.

**Reply by Build-0-Matic** [reply]
Yes, that works!
still, I wonder what the -format png32 does that photoshop doesn't like?

**Comment by snibgo** [comment]
I don't use Photoshop, and can't advise on that.

I don't suppose "`-format png24`" (or 32) makes any difference to your command. See the documentation at https://imagemagick.org/script/command-line-options.php#format

When used with "magick mogrify", "`-format`" determines the output image file format. But in "magick convert", or simply "magick", it determines the format of the output "info:" text.

Incidentally, as you are using IM v7, I suggest using just "magick", not "magick convert". The difference is whether the command takes v7 or v6 syntax. V7 is much better.

**Reply by Build-0-Matic** [reply]
Well, I learned something!
Thanks for the tips and the explanation.
